src/App.tsx

```diff
@@ -1,28 +1,75 @@
-import { useState } from "react";
-import { getPublicKey } from "./tauriCommands";
+import { useEffect, useRef, useState } from "react";
+
+import { Button } from "./components/ui/button";
+import {
+  Dialog,
+  DialogContent,
+  DialogFooter,
+  DialogHeader,
+  DialogTitle,
+} from "./components/ui/dialog";
+import { handleSignEventRequests } from "./tauriCommands";
 
 const App = () => {
   const [publicKey, setPublicKey] = useState("");
+  const [open, setOpen] = useState(false);
+  const [event, setEvent] = useState({});
+  const resolveRejectRef = useRef<{
+    resolve: (value: boolean) => void;
+    reject: (value: boolean) => void;
+  } | null>(null);
+
+  useEffect(() => {
+    const handleEvent = (event: any): Promise<boolean> => {
+      setEvent(event);
+      setOpen(true);
+      return new Promise((resolve, reject) => {
+        resolveRejectRef.current = { resolve, reject };
+      });
+    };
+
+    handleSignEventRequests(handleEvent);
```

Comment:
`handleSignEventRequests()` returns a function that unregisters the event handler (`handleEvent`). I know this doesn't really matter (yet) since this component won't ever unrender, but for best practice can we return this event handler unregister function here so that _if_ this ever gets unrendered by React, it removes the hook?

I think all we need to do is add a `return`:

`return handleSignEventRequests(handleEvent);`